### DISCUSSION TITLE
sourcemap.json for custom(none rojo based) projects

### DISCUSSION THREAD

**Discussion by mundusnine** [JohnnyMorganz/luau-lsp#320] [Q&A] [answered]
So I am using luau and trying to add definition files for the objects added to my runtime with my native code. It is said in the repo readme that we can use sourcemap.json files in these situations. The issue I am having is I can't seem to actually make them work with luau-lsp. I based my sourcemap.json format on [this](https://github.com/filiptibell/rojo-sourcemap-action/blob/main/test/bin/build.project.json) 

And mine looks like this: 
```json
{
  "name": "luau-project",
  "tree": {
      "Project": {
        "$className": "Project",
        "$path": "../project.d.luau" // I also tried .. so that it resolves the files by itself, but no cigar
      }
  }
}
```
The behavior I am getting is , the lsp server looks for default.project.json and fails because I am not using rojo. It never seems to look  for the sourcemap.json or if it finds it, it doesn't seem to update with my definitions. I tested my definitions with the extension and they work(by specifying their path in vscode directly) 

Did anyone test this or got it to work ? Does anyone have any clue what I am doing wrong ?

Thank you and great work on luau-lsp, when it works it's greaaat !

**Comment by JohnnyMorganz** [comment, marked as answer]
I assume this is for a non-Roblox project when you say "added to my runtime with my native code"?

If so, ignore the Rojo, default.project.json, and sourcemap part. They are only for Roblox projects (to allow resolving stuff like `require(game.ReplicatedStorage.Module)`.

For standalone Luau projects, you should just load the definitions files directly in VSCode through the setting, like you have already done so. Requires are then done by string (and can be configured to either be relative to current file or workspace root).

Is there a reason you are looking at sourcemap rather than the VSCode setting?

**Reply by mundusnine** [reply]
Yes, basically if I share my definitions with other people it would help to have a sourcemap file to enable configuring what luau-lsp see's without having to add each and every file manually to vscode. Manually adding them with the UI is also quite clunky and if I change computer(or OS) I don't want to reconfigure for each definition file created. Basically, in the native world you are creating a lot of luau objects and you want to have a way for luau-lsp to know that these types exist even if they are created directly in the runtime.

This would also add the benefit of reuse for other implementers/integrators of luau-lsp to other editors.

**Reply by JohnnyMorganz** [reply]
One thing I would recommend is that you configure the setting at the workspace level rather than the user level in your settings. This will then create a `.vscode/settings.json` file in your workspace which can list these definition files specifically for your project. Hopefully that can solve your needs
